drivers/adc/adc_nrfx_saadc.c

```diff
@@ -358,6 +360,20 @@ static void saadc_irq_handler(void *param)
 {
 	struct device *dev = (struct device *)param;
 
+	if (nrf_saadc_event_check(NRF_SAADC_EVENT_CALIBRATEDONE)) {
+		nrf_saadc_event_clear(NRF_SAADC_EVENT_CALIBRATEDONE);
+		nrf_saadc_event_clear(NRF_SAADC_EVENT_STOPPED);
+		nrf_saadc_task_trigger(NRF_SAADC_TASK_STOP);
+		while (!nrf_saadc_event_check(NRF_SAADC_EVENT_STOPPED)) {
```

Comment:
I don't entirely understand the purpose of this part of code. Why can't you just start the sampling after getting (and clearing) the CALIBRATEDONE event?
Such loop in the interrupt handler does not look nice.

Reply:
Oh, I see it now, PAN-86. A word of comment here could help.